stdlib/src/python/_cpython.mojo

```diff
@@ -1009,6 +1009,26 @@ struct CPython:
         self._inc_total_rc()
         return r
 
+    fn PyObject_SetItem(
+        inout self, obj: PyObjectPtr, key: PyObjectPtr, value: PyObjectPtr
+    ) -> Int:
+        var r = self.lib.get_function[
+            fn (PyObjectPtr, PyObjectPtr, PyObjectPtr) -> Int
```

Comment:
```suggestion
            fn (PyObjectPtr, PyObjectPtr, PyObjectPtr) -> c_int
```

Otherwise, this is incorrect from an ABI perspective (Int is 8 byte but the function in CPython is `int`, which is 4 bytes.  If you'd like to fix this here and for the other function pointers that run afoul of this problem, that would be great! 

These are incorrectly using Int, which is 64 bit, whereas c_int is almost always 32 bits on modern platforms we support.

This hasn't been a major problem, particularly for return values, because Int is typically a small-valued error status code stored in a little-endian register upon return.

Reply:
Thank you for making me aware of this. Will update this addition and open a separate PR for the others

Reply:
That would be very much appreciated, thank you!  We're looking all the rough edges of FFI right now, including the ABI struggles, as we've been working in this neighborhood with CPython APIs recently (@ConnorGray has gotten burned a time or two I'd say lately 😆 ).

Reply:
I think we should put a link to the Python doc in all these functions in `CPython` so people can check for API/ABI mismatch more easily.
```mojo
# Ref: https://docs.python.org/3/c-api/object.html#c.PyObject_SetItem
```

Reply:
> Regarding consistency, is there any preference for matching the return types? ...
> ... put a link to the Python doc in all these functions in `CPython` ...
@jjvraw Let's create an issue to track these so Conner can let us know what's the best way forward.

Reply:
I'm actually of the opinion that in this case, there is no benefit in the type cast. At least for how `PyObject_IsTrue` is currently being used in the lib now.

However, given the points Joe made and that is seems to be an internal goal at the moment, it might be better for Joe, or someone from the team, to open an issue